### PR TITLE
feat(eval): add parse_promql_duration comparison function

### PR DESCRIPTION
**Pull request by MichaHoffmann**
The approach to parse a duration is a bit gross but seemed to work well enough in tests. The function name is debatable since this is only interesting in duration metavariables captured from promql i think.

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [ ] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

**Comment by aryx** [comment]
Until now we've tried in Eval_generic to use the name of Python functions, as most people are familiar with Python so it's a good starting point. Dunno what's the corresponding function in Python for this "milliseconds" but we can always refine that later in a separate PR.

**Comment by aryx** [comment]
At some point you also need to make a PR for our semgrep-docs so we update also:
https://semgrep.dev/docs/writing-rules/rule-syntax/#metavariable-comparison

**Comment by aryx** [comment]
Please add a changelog.d/ entry for this. Every user-facing improvement should be added in the changelog so users can learn about it.

**Comment by MichaHoffmann** [comment]
> Until now we've tried in Eval_generic to use the name of Python functions, as most people are familiar with Python so it's a good starting point. Dunno what's the corresponding function in Python for this "milliseconds" but we can always refine that later in a separate PR.

there is no corresponding python function really in this case! its also only useful for promql durations.

**Comment by aryx** [comment]
See https://github.com/returntocorp/semgrep/pull/8053 for example of changelog.d entry

File: src/engine/Eval_generic.ml [hunk 1]
```diff
@@ -236,6 +278,13 @@ let rec eval env code =
       | __else__ -> raise (NotHandled code))
   | G.Call ({ e = G.N (G.Id (("today", _), _)); _ }, (_, _, _)) ->
       Float (Unix.time ())
+  (* Convert duration string to integer milliseconds*)
```
Comment (aryx) [inline src/engine/Eval_generic.ml]: maybe add a comment then here saying there's no equivalent python function so we use the name used
in prometheus (with a link to the prometheus doc).

Comment (MichaHoffmann) [reply src/engine/Eval_generic.ml]: i dont think there is a corresponding go function too ( or maybe `time.ParseDuration(...).Milliseconds()` which is a bit of a mouthfull)

**Review by aryx — APPROVED** [review]

**Pull request merged**